backend/authentication-service/app/middlewares/authJwt.js

```diff
@@ -0,0 +1,42 @@
+const jwt = require("jsonwebtoken");
+const config = require("../config/auth.config.js");
+const db = require("../models");
+const User = db.user;
+const Role = db.role;
+
+verifyToken = (req,res,next) => {
+    let token = req.session.token;
+
+    if (!token) {
+        return res.status(403).send({message:"No token provided!"});
+    }
+    
+    jwt.verify(token, config.secret, (err,decoded) => {
+        if(err) {
+            return res.status(401).send({message:"sir stop sir, u are unauthorized!"});
+        }
+        req.userId = decoded.id;
+        next();
+    });
+};
+
+isAdmin = (req,res,next) => {
+    User.findById(req.userId).exec().then((user) => {
+        Role.find({_id:{$in:user.roles}}).then((roles) => {
+            for(let i=0; i<roles.length; i++) {
+                if(roles[i].name === "admin") {
+                    next();
+                    return;
+                }
+            }
+            res.status(403).send({message:"sir stop sir, you require admin role!"});
```

Comment:
Error message should be changed, possibly in a future PR. Similar cases in other parts of the file